libaftermath-core/src/defs/aftermath/types/openmp/on_disk.py

```diff
@@ -238,14 +239,471 @@
+            name = "prior_task_status",
+            field_type = aftermath.types.builtin.uint8_t,
+            comment = "Status of the task at the scheduling point")]))
+
+tags.dsk.tomem.add_per_event_collection_tags(
+    am_dsk_openmp_task_schedule,
+    aftermath.types.openmp.in_memory.am_openmp_task_schedule,
+    "collection_id")
+
+################################################################################
+
+am_dsk_openmp_implicit_task = EventFrame(
+    name = "am_dsk_openmp_implicit_task",
+    entity = "on-disk OpenMP implicit task execution interval",
+    comment = "An OpenMP implicit task execution interval",
+    fields = FieldList([
+        Field(
+            name = "interval",
+            field_type = aftermath.types.on_disk.am_dsk_interval,
+            comment = "Start and end of the execution interval"),
+        Field(
+            name = "actual_parallelism",
+            field_type = aftermath.types.builtin.uint32_t,
+            comment = "Number of threads or teams in the region"),
+        Field(
+            name = "index",
+            field_type = aftermath.types.builtin.uint32_t,
+            comment = "Thread or team number of the calling thread"),
+        Field(
+            name = "flags",
+            field_type = aftermath.types.builtin.uint8_t,
+            comment = "Indication if the task is initial or implicit")]))
+
+tags.dsk.tomem.add_per_event_collection_tags(
+    am_dsk_openmp_implicit_task,
+    aftermath.types.openmp.in_memory.am_openmp_implicit_task,
+    "collection_id")
+
+################################################################################
+
+am_dsk_openmp_sync_region_wait = EventFrame(
+    name = "am_dsk_openmp_sync_region_wait",
+    entity = "on-disk OpenMP sync region wait execution interval",
+    comment = "An OpenMP sync region wait execution interval",
+    fields = FieldList([
+        Field(
+            name = "interval",
+            field_type = aftermath.types.on_disk.am_dsk_interval,
+            comment = "Start and end of the execution interval"),
+        Field(
+            name = "kind",
+            field_type = aftermath.types.builtin.uint8_t,
+            comment = "Argument indicates a type of the synchronization "
+                      + "construct")]))
+
+tags.dsk.tomem.add_per_event_collection_tags(
+    am_dsk_openmp_sync_region_wait,
+    aftermath.types.openmp.in_memory.am_openmp_sync_region_wait,
+    "collection_id")
+
+################################################################################
+
+am_dsk_openmp_mutex_released = EventFrame(
+    name = "am_dsk_openmp_mutex_released",
+    entity = "on-disk OpenMP mutex released event",
+    comment = "An event generated when the OpenMP mutex is released",
+    fields = FieldList([
+        Field(
+            name = "timestamp",
+            field_type = aftermath.types.builtin.uint64_t,
+            comment = "Time of the event"),
+        Field(
+            name = "wait_id",
```

Comment:
I agree with that let's keep it as a future extension. To avoid overhead in tracing it would probably derived information in the some post-processing phase.